scripts/document_indexer.py

```diff
@@ -0,0 +1,315 @@
+import argparse
+import csv
+import json
+import os
+import pickle
+import re
+import traceback
+from typing import Dict, List, Literal, Optional, Tuple
+
+from tiktoken import Encoding
+
+try:
+    import docx2txt
+    import pptx
+    import tiktoken
+    from bs4 import BeautifulSoup
+    from langchain_community.embeddings import HuggingFaceEmbeddings
+    from langchain_community.vectorstores import FAISS
+    from pypdf import PdfReader
```

Comment:
could we make those dependencies optional and only load them when there are specific types of documents?

Reply:
done